test/tools/ossfuzz/protoToAbiV2.cpp

```diff
@@ -320,38 +372,82 @@ string ProtoConverter::visit(TestFunction const& _x, string const& _storageVarDe
 	string localVarDefs = localVarBuffers.second;
 
 	ostringstream testBuffer;
-	string functionDecl = "function test() public returns (uint)";
+
+	string testFunction = Whiskers(R"(
+	function test() public returns (uint) {
+		<?calldata>return test_calldata_coding();</calldata>
+		<?returndata>return test_returndata_coding();</returndata>
+	})")
+	("calldata", m_test == Contract_Test::Contract_Test_CALLDATA_CODER)
+	("returndata", m_test == Contract_Test::Contract_Test_RETURNDATA_CODER)
+	.render();
+
+	string functionDeclCalldata = "function test_calldata_coding() internal returns (uint)";
+	string functionDeclReturndata = "function test_returndata_coding() internal returns (uint)";
+
 	testBuffer << Whiskers(R"(<structTypeDecl>
-	<functionDecl> {
+<testFunction>
+<?calldata>
+	<functionDeclCalldata> {
 <storageVarDefs>
 <localVarDefs>
-<testCode>
-	})")
+<calldataTestCode>
+	}
+<calldataHelperFuncs>
+</calldata>
+<?returndata>
+	<functionDeclReturndata> {
+<returndataTestCode>
+	}
+
+<?varsPresent>
+	function coder_returndata_public() public returns (<return_types>) {
```

Comment:
I don't think you actually need the public version - return values are only ABI encoded when calling externally anyways - and calling two otherwise equal public and external function externally doesn't make a difference (except for their arguments, but there aren't any here anyways).

Reply:
Ah I see, thank you